src/common.h

```diff
@@ -25,6 +25,18 @@
 
 #define ARRAYS_SIZE(arr) (sizeof(arr) / sizeof(arr[0]))
 
+#define MASK(n) (~((~0U << (n))))
+
+/* Floor log base 2 */
```

Comment:
Reimplement `ilog2` with static inline function, allowing type checking. In addition, there should be a generic C implementation.